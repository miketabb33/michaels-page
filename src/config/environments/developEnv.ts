import { EnvConfig } from './currentEnv'

const firebaseConfig_Dev = {
  apiKey: 'AIzaSyDprPwOCjEwEYcbW4nuPxFWtEOJFMc6bek',
  authDomain: 'michael-website-19ed0.firebaseapp.com',
  projectId: 'michael-website-19ed0',
  storageBucket: 'michael-website-19ed0.appspot.com',
  messagingSenderId: '855765427298',
  appId: '1:855765427298:web:763f3885091a33b5231d3b',
  measurementId: 'G-PP96RS59JC',
}

export const developEnv: EnvConfig = {
  firebaseConfig: firebaseConfig_Dev,
  id: 'develop',
}
