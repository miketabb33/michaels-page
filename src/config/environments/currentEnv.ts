import { FirebaseOptions } from 'firebase/app'
import { productionEnv } from './productionEnv'
import { developEnv } from './developEnv'

export type EnvConfig = {
  firebaseConfig: FirebaseOptions
}

export const getEnv = (siteEnv: string): EnvConfig => {
  if (siteEnv === 'production') return productionEnv
  else return developEnv
}

export const ENV: EnvConfig = getEnv(SITE_ENV)
