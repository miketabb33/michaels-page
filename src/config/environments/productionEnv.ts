import { EnvConfig } from './currentEnv'

const firebaseConfig_Prod = {
  apiKey: 'AIzaSyCtqRREY6znnMH19OieyldddLfkT6tftqk',
  authDomain: 'michael-website-prod.firebaseapp.com',
  projectId: 'michael-website-prod',
  storageBucket: 'michael-website-prod.appspot.com',
  messagingSenderId: '798894266083',
  appId: '1:798894266083:web:9688813c4e93d2dc7cd7d6',
  measurementId: 'G-JN9MYTGX8R',
}

export const productionEnv: EnvConfig = {
  firebaseConfig: firebaseConfig_Prod,
}
