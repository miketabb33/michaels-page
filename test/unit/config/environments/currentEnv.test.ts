import { getEnv } from '../../../../src/config/environments/currentEnv'

describe('Get Env', () => {
  it('should match snapshot for production env', () => {
    expect(getEnv('production')).toMatchSnapshot()
  })
  it('should match snapshot for other env', () => {
    expect(getEnv('develop')).toMatchSnapshot()
  })
})
