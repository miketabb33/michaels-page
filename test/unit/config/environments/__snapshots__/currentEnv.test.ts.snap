// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Env should match snapshot for other env 1`] = `
{
  "firebaseConfig": {
    "apiKey": "AIzaSyDprPwOCjEwEYcbW4nuPxFWtEOJFMc6bek",
    "appId": "1:855765427298:web:763f3885091a33b5231d3b",
    "authDomain": "michael-website-19ed0.firebaseapp.com",
    "measurementId": "G-PP96RS59JC",
    "messagingSenderId": "855765427298",
    "projectId": "michael-website-19ed0",
    "storageBucket": "michael-website-19ed0.appspot.com",
  },
}
`;

exports[`Get Env should match snapshot for production env 1`] = `
{
  "firebaseConfig": {
    "apiKey": "AIzaSyCtqRREY6znnMH19OieyldddLfkT6tftqk",
    "appId": "1:798894266083:web:9688813c4e93d2dc7cd7d6",
    "authDomain": "michael-website-prod.firebaseapp.com",
    "measurementId": "G-JN9MYTGX8R",
    "messagingSenderId": "798894266083",
    "projectId": "michael-website-prod",
    "storageBucket": "michael-website-prod.appspot.com",
  },
}
`;
